### PR TITLE
Deguice SearchService and friends

### PR DESCRIPTION
**Pull request by s1monw**
This change removes the guice dependency handling for SearchService and
several related classes like SearchTransportController and SearchPhaseController.
The latter two now have pkg private constructors and dependencies like FetchPhase
are now created by calling their ctors explicitly. This also cleans up several users
of the DefaultSearchContext and centralized it's creation inside SearchService.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/get/GetRequest.java [hunk 1]
```diff
@@ -61,7 +61,6 @@
 
     private VersionType versionType = VersionType.INTERNAL;
     private long version = Versions.MATCH_ANY;
-    private boolean ignoreErrorsOnGeneratedFields;
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/action/get/GetRequest.java]: Note to any other reviewers: this is no longer supported. I haven't dug too deep but my guess is that it was removed when we removed GET from the translog.

File: core/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java [hunk 2]
```diff
@@ -162,16 +160,16 @@ void performFirstPhase(final int shardIndex, final ShardIterator shardIt, final
...
-                        public void onResponse(FirstResult result) {
-                            onFirstPhaseResult(shardIndex, shard, result, shardIt);
-                        }
-
-                        @Override
-                        public void onFailure(Exception t) {
-                            onFirstPhaseResult(shardIndex, shard, node.getId(), shardIt, t);
-                        }
-                    });
+                    @Override
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java]: The old indentation was better I think.

File: core/src/main/java/org/elasticsearch/action/get/GetRequest.java (hunk 1 above)
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/action/get/GetRequest.java]: correct

File: core/src/test/java/org/elasticsearch/search/SearchRequestTests.java [hunk 3]
```diff
@@ -56,12 +56,7 @@ protected void configure() {
             }
         };
         SearchModule searchModule = new SearchModule(Settings.EMPTY, false,
-                Collections.singletonList(new FetchSubPhasePluginIT.FetchTermVectorsPlugin())) {
-            @Override
-            protected void configureSearch() {
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/search/SearchRequestTests.java]: Nice to see these go.

**Comment by nik9000** [comment]
LGTM

File: core/src/main/java/org/elasticsearch/node/Node.java [hunk 4]
```diff
@@ -795,8 +789,10 @@ BigArrays createBigArrays(Settings settings, CircuitBreakerService circuitBreake
     /**
      * Select the search service implementation. Overrided by tests.
      */
-    protected Class<? extends SearchService> pickSearchServiceImplementation() {
-        return SearchService.class;
+    protected SearchService newSearchService(ClusterService clusterService, IndicesService indicesService,
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/node/Node.java]: This is exactly what I had envisioned. Thank you!

**Comment by rjernst** [comment]
Great! LGTM too.

**Pull request merged**